app/models/settings.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class Settings < ApplicationRecord
+  validates :singleton_guard, inclusion: { in: [0] }
```

Comment:
```suggestion
  validates :singleton_guard, inclusion: { in: [0] }, uniqueness: true
```

I think you'll want this as well, sort of a guard for future folks working on this in case they forget to use `instance`... Also it just matches the functionality you're going for.